index.html

```diff
@@ -140,67 +166,250 @@ <h2>
+          <a data-cite="html#responsible-document">responsible document</a> to
+          <a>nothing</a>, and return a <a data-cite=
+          "webidl#a-promise-resolved-with">resolved promise</a>.
+        </p>
+      </section>
+      <section>
+        <h3>
+          <dfn data-dfn-for="NavigatorBadge">setAppBadge()</dfn> method
+        </h3>
+        <p>
+          When the <code>setAppBadge()</code> method is called with argument
+          <var>contents</var>, run the following steps:
+        </p>
+        <ol>
+          <li>Let <var>result</var> be the result of <a>determining the
+          matching applications</a>. If <var>result</var> was a
+          {{DOMException}}, return a <a data-cite=
+          "webidl#a-promise-rejected-with">rejected promise</a> with reason
+          <var>result</var>.
+          </li>
+          <li>For each <var>app</var> in <var>result</var>:
+            <ol>
+              <li>If <var>contents</var> is omitted, set the badge associated
+              with <var>app</var> to <a>flag</a>.
+              </li>
+              <li>Else, if <var>contents</var> is 0, set the badge associated
+              with <var>app</var> to <a>nothing</a>.
+              </li>
+              <li>Else, set the badge associated with <var>app</var> to
+              <var>contents</var>.
+              </li>
+            </ol>
+          </li>
+          <li>Return a <a data-cite="webidl#a-promise-resolved-with">resolved
+          promise</a>.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          <dfn data-dfn-for="NavigatorBadge">clearAppBadge()</dfn> method
+        </h3>
+        <p>
+          When the <code>clearAppBadge()</code> method is called:
+        </p>
+        <ol>
+          <li>Let <var>result</var> be the result of <a>determining the
+          matching applications</a>. If <var>result</var> was a
+          {{DOMException}}, return a <a data-cite=
+          "webidl#a-promise-rejected-with">rejected promise</a> with reason
+          <var>result</var>.
+          </li>
+          <li>For each <var>app</var> in <var>result</var>:
+            <ol>
+              <li>Set the badge associated with <var>app</var> to
+              <a>nothing</a>.
+              </li>
+            </ol>
+          </li>
+          <li>Return a <a data-cite="webidl#a-promise-resolved-with">resolved
+          promise</a>.
           </li>
         </ol>
-        <div class="issue">
-          It is a bit weird that <code>set(0)</code> clears the badge, while
-          <code>set()</code> sets it to <a>flag</a>. This is designed so that
-          the above example code for setting the unread count will
-          automatically clear the badge when the unread count is 0. But it
-          could be a bit unintuitive.
-        </div>
       </section>
       <section>
         <h3>
-          <dfn>clear()</dfn> function
+          Determining the set of matching applications
```

Comment:
Should this be above the definition of `setAppBadge` or below `clearAppBadge`. Seems kind of odd to have it in the middle.